### PR TITLE
Burak-last-hashtag-redirect

### PR DESCRIPTION
**Pull request by myniqx**
https://github.com/freedomcombination/monorepo/issues/1155

### CODE REVIEW

File: apps/trend-rights/middleware.ts [hunk 1]
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
   'Access-Control-Allow-Headers': 'Content-Type, Authorization',
 }
 
-export function middleware(request: NextRequest) {
-  if (request.method === 'OPTIONS') {
-    return NextResponse.json({}, { headers: corsHeaders })
+const dontRedirectIfDevelopment = true // prevents redirecting if in development
```
Comment (7alip) [inline apps/trend-rights/middleware.ts:11]: Let's redirect in dev mode as well. We can easily check it always

File: apps/trend-rights/middleware.ts [hunk 2]
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
...
+  const hashtagDatetimePlus12Hours = addHours(hashtagDatetime, 12)
+
+  const currentDate = new Date()
+  const isCurrentDateWithinInterval = isWithinInterval(currentDate, {
+    start: hashtagDatetime,
+    end: hashtagDatetimePlus12Hours,
+  })
+
+  if (isCurrentDateWithinInterval) {
+    return NextResponse.redirect('/hashtags/' + slug)
```
Comment (7alip) [inline apps/trend-rights/middleware.ts:47]: https://nextjs.org/docs/pages/building-your-application/routing/redirecting#nextresponseredirect-in-middleware

In Next.js pages router example, they use redirect like this:

`return NextResponse.redirect(new URL('/login', request.url))`

File: apps/trend-rights/middleware.ts [hunk 3]
```diff
@@ -6,18 +8,42 @@ const corsHeaders = {
...
+  const hashtagDatetime = parseISO(dateStr)
+  const hashtagDatetimePlus12Hours = addHours(hashtagDatetime, 12)
+
+  const currentDate = new Date()
+  const isCurrentDateWithinInterval = isWithinInterval(currentDate, {
+    start: hashtagDatetime,
+    end: hashtagDatetimePlus12Hours,
+  })
+
+  if (isCurrentDateWithinInterval) {
```
Comment (7alip) [inline apps/trend-rights/middleware.ts:46]: Check if redirected already

```ts
const isRedirected = slug === cookieSlug
```

**Review by 7alip — COMMENTED** [review]
Thanks

**Comment by myniqx** [comment]
@7alip im sorry i didnt get review request back. i rewrote almost everthing. its better.
what is working:
- reads from edge-config 
- if it tries to fetch the key more than once in 10 seconds, it skips (cookie named dofetch)
- if it already redirected (cookie named redirectedslug) or already in hashtags/:slug, it skips.
- in backend afterCreate and afterUpdate without check any condition last hashtag ,in teory, added to edge config.
what is not working:
-last hashtag cannot be updated to edge config.

i test it under there condition:
- i return static last-hashtag and do testing and get those result (above)
- and also backend works until set function. it returns an error 403. it seems that token in connection string is working while reading, but when create or update, it doesnt work.
if you paste the connection string you gave me to address bar you can easly see its working.
but when write (https://vercel.com/docs/storage/edge-config/vercel-api#update-your-edge-config-items)
this is the exact same error : 
```
{
  "error": {
    "code": "forbidden",
    "message": "The request is missing an authentication token",
    "missingToken": true
  }
}
```
maybe with the correct token, this ll work.

**Comment by myniqx** [comment]
@7alip now its better :)
in middleware:
i added matcher, with a litlle bir more filter it works without do-fetch cookie. so i remove it.
vercel's get function is now working. it seems it was a matter releated with turbo. 
and i added another edge key for development mode, 'last-hashtag-dev'.

**Review by 7alip — APPROVED** [review]
Thanks

**Pull request merged**